moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -200,11 +276,57 @@ const NewProjectMap = ({
       : "default";
   };
 
+  /**
+   * Whenever a Save Action is initiated, save all drawn features
+   * */
+  useEffect(() => {
+    // Ignore empty states
+    if (!saveActionState) return;
+    // If the process has been already initiated, we don't need to go any further
+    if (saveActionState?.currentStep && saveActionState.currentStep > 1) return;
+    // It looks like this is the first step
+    if (
+      saveActionState?.initiateFeatureSave &&
+      saveActionState?.featuresSaved === false
+    ) {
+      saveDrawnPoints();
+    }
+  }, [saveActionState, saveDrawnPoints]);
+
   return (
     <Box className={noPadding ? classes.mapBoxNoPadding : classes.mapBox}>
-      {/* Render these controls outside ReactMapGL so mouse events don't propagate to the map */}
-      <div ref={mapControlContainerRef} className={classes.geocoderContainer} />
-      {renderLayerSelect()}
+      <style>
+        {!visibleLayerIds.includes("drawnByUser") &&
+          `g > circle { display: none; }`}
+        {!visibleLayerIds.includes("drawnByUserLine") &&
+          `g > path { display: none; }`}
+      </style>
+
+      {/* The following div acts as an anchor and it specifies where the geocoder will live */}
+      <div
+        ref={mapGeocoderContainerRef}
+        className={classes.geocoderContainer}
+      />
+      <div
+        ref={mapEditToolsContainerRef}
+        className={classes.mapBoxEditButtonGroup}
+      />
+      <div ref={mapBasemapContainerRef} className={classes.speedDial} />
```

Comment:
In these past few lines, the divs provide a landing area for our portals. Notice they are outside the map.